File: c-code/shapeJudgeModule/PointInfo.cpp
// PointInfo.cpp: implementation of the PointInfo class.
//
//////////////////////////////////////////////////////////////////////

#include "stdafx.h"
#include "PointInfo.h"

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

PointInfo::PointInfo(int id,double lo,double la,short speed,short dir,time_t time)
{
	this->id = id;
	this->lo = lo;
	this->la = la;
	this->speed = speed;
	this->dir = dir;
	this->time = time;

//	printf("mdtId=%d; time=%dl; lo=%f; la=%f; speed=%d; dir=%d; \n",id,time,lo,la,speed,dir);
}

PointInfo::~PointInfo()
{
	
}

int PointInfo::getId(){
	return this->id;
}
char* PointInfo::getDesc(){
	return this->desc;
}
double PointInfo::getLo(){
	return this->lo;
}
double PointInfo::getLa(){
	return this->la;
}
short PointInfo::getSpeed(){
	return this->speed;
}
short PointInfo::getDir(){
	return this->dir;
}
time_t PointInfo::getTime(){
	return this->time;
}

void PointInfo::setAssociateInfo(int segmentId,int pointIndex,long lom,long lam,long minDistance,int segmentOrder){
	this->segmentId = segmentId;
	this->pointIndex = pointIndex;
	this->assLom = lom;
	this->assLam = lam;
	this->distance = minDistance;
	this->segmentOrder = segmentOrder;
}
int PointInfo::getSegmentId(){
	return this->segmentId;
}
int PointInfo::getSegmentOrder(){
	return this->segmentOrder;
}
int PointInfo::getPointIndex(){
	return this->pointIndex;
}
long PointInfo::getAssLom(){
	return this->assLom;
}
long PointInfo::getAssLam(){
	return this->assLam;
}
long PointInfo::getDistance(){
	return this->distance;
}

File: src/com/fleety/base/shape/java/PointInfo.java
package com.fleety.base.shape.java;

public class PointInfo {
	private int id = 0;
	private String desc;
	private double lo;
	private double la;
	private short dir;

	private int segmentId;
	private int segmentOrder = 0;
	private int pointIndex = 0;
	private int assLom = 0;
	private int assLam = 0;
	private int distance = 0;
	
	public PointInfo(int id,double lo,double la,short dir){
		this.id = id;
		this.lo = lo;
		this.la = la;
		this.dir = dir;
	}
	
	public int getId(){
		return this.id;
	}
	public String getDesc(){
		return this.desc;
	}
	public double getLo(){
		return this.lo;
	}
	public double getLa(){
		return this.la;
	}
	public short getDir(){
		return this.dir;
	}

	public void setAssociateInfo(int segmentId,int pointIndex,int lom,int lam,int distance){
		this.setAssociateInfo(segmentId, pointIndex, lom, lam, distance, 1);
	}
	public void setAssociateInfo(int segmentId,int pointIndex,int lom,int lam,int distance,int segmentOrder){
		this.segmentId = segmentId;
		this.pointIndex = pointIndex;
		this.assLom = lom;
		this.assLam = lam;
		this.distance = distance;
		this.segmentOrder = segmentOrder;
	}
	public int getSegmentId(){
		return this.segmentId;
	}
	public int getSegmentOrder(){
		return this.segmentOrder;
	}
	public int getPointIndex(){
		return this.pointIndex;
	}
	public int getAssLom(){
		return this.assLom;
	}
	public int getAssLam(){
		return this.assLam;
	}
	public int getDistance(){
		return this.distance;
	}
}
